docker/README.md

```diff
@@ -0,0 +1,254 @@
+- For a pipeline running in Jenkins X, use [Skaffold](#with-skaffold).
+- In any case, you can use [Docker](#with-docker).
+
+### With Maven
+
+To build all the images locally, run:
+
+```
+mvn -nsu install
+```
+
+To build a single image, for instance the `nuxeo/builder` one, run:
+
+```
+mvn -nsu -f builder/pom.xml install
+```
+
+### With Skaffold
+ 
+We use Skaffold to build the images as part of the [nuxeo](http://jenkins.platform.34.74.59.50.nip.io/job/nuxeo/job/nuxeo/) pipeline in our Jenkins X CI/CD platform.
+
+This requires to:
+- Install [Skaffold](https://skaffold.dev/docs/getting-started/#installing-skaffold).
+- Install [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/).
+- Configure `kubectl` to connect to a Kubernetes cluster.
+
+It also requires the following environment variables:
+- `DOCKER_REGISTRY`: the Docker registry to push the images to.
+- `ORG`: the organization used as a namespace in the image names, in our case: `nuxeo`.
+- `VERSION`: the image tag, for instance `11.1-SNAPSHOT`.
+
+Because we are using the [Kaniko](https://github.com/GoogleContainerTools/kaniko) builder and there is no support (yet) for [ordering dependant builds](https://github.com/GoogleContainerTools/skaffold/issues/2717), we are relying on several `skaffold.yaml` files and performing a sequential build, see the `skaffoldBuildAll` method in the [Jenkinsfile](../Jenkinsfile).
+
+First, to build the `nuxeo/builder` and `nuxeo/base` images, run:
+
+```
+skaffold build
+```
+
+To build the `nuxeo/builder-platform` image, run:
+
+```
+skaffold build -f builder-platform/skaffold.yaml
+```
+
+To build the `nuxeo/nuxeo` image, run:
+
+```
+skaffold build -f slim/skaffold.yaml
+```
+
+### With Docker
+
+To build the `nuxeo/builder` image, you first need to fetch the Nuxeo distribution and make it available for the Docker build with Maven:
+
+```
+mvn -nsu -f builder/pom.xml process-resources
+```
+
+Then run:
+
+```
+docker build -t nuxeo/builder:11.1-SNAPSHOT -f builder/Dockerfile builder
+```
+
+To build the `nuxeo/base` image, run:
+
+```
+docker build -t nuxeo/base:11.1-SNAPSHOT -f base/Dockerfile base
+```
+
+To build the `nuxeo/builder-platform` image, you first need to fetch the Nuxeo packages to install and make them available for the Docker build with Maven:
+
+```
+mvn -nsu -f builder-platform/pom.xml process-resources
+```
+
+Then run:
+
+```
+docker build -t nuxeo/builder-platform:11.1-SNAPSHOT -f builder-platform/Dockerfile --build-arg BASE_IMAGE=nuxeo/builder:11.1-SNAPSHOT builder-platform
+```
+
+To build the `nuxeo/nuxeo` image, run:
+
+```
+docker build -t nuxeo/nuxeo:11.1-SNAPSHOT -f slim/Dockerfile --build-arg BUILDER_IMAGE=nuxeo/builder:11.1-SNAPSHOT --build-arg BASE_IMAGE=nuxeo/base:11.1-SNAPSHOT slim
+```
+
+## Run an Image
+
+For instance, to run a container derived from the the `nuxeo/nuxeo` image built locally, run:
+
+```
+docker run -it -p 8080:8080 nuxeo/nuxeo:11.1-SNAPSHOT
```

Comment:
here is the example of a direct conflict with the official docker hub image which is also nuxeo/nuxeo

Reply:
In the image name yes, bu the tag should be discriminant: 11.1-SNAPSHOT vs latest.
Again, once we decide what deployment policy regarding these images vs the official ones we might adapt the image names.